src/deluge/dsp/filter/filter_set.h

```diff
@@ -59,12 +60,17 @@ class FilterSet {
 	void renderLPFLongStereo(q31_t* startSample, q31_t* endSample);
 	void renderHPFLongStereo(q31_t* startSample, q31_t* endSample);
 	void renderHPFLong(q31_t* startSample, q31_t* endSample, int32_t sampleIncrement = 1);
-	void renderLadderHPF(q31_t* outputSample);
 
-	SVFilter lpsvf;
-	LpLadderFilter lpladder;
-	HpLadderFilter hpladder;
-	SVFilter hpsvf;
+	// all filters share a state. This is fine since they just hold plain data and initialization is handled by
+	// reset/configure calls.  This is faster than using a variant at the cost of not throwing on unitialized access.
+	// However since there are no invariants to uphold, the worst case scenario is an audio glitch so whatever
+	// This cuts 250 bytes from the size of this class, which is fairly significant since every voice has a filterset
+	char lpfilterstate[lpfilterSize];
+	char hpfilterstate[hpfilterSize];
+	SVFilter& lpsvf = *(SVFilter*)lpfilterstate;
+	LpLadderFilter& lpladder = *(LpLadderFilter*)lpfilterstate;
+	HpLadderFilter& hpladder = *(HpLadderFilter*)hpfilterstate;
+	SVFilter& hpsvf = *(SVFilter*)hpfilterstate;
```

Comment:
Can we use a `union` instead? Same no-throw behavior but without the questionable casting, I think.

Reply:
Smart - I thought you could only use POD structs in unions but looks like that may not be the case anymore